src/components/UserInputFormArea.tsx

```diff
@@ -32,31 +52,51 @@ const UserInputFormArea: React.FC = () => {
     };
 
     useEffect(() => {
-        if (!departureStation || !destinationStation) {
-            setMessage('Please select both stations');
+
+        const selectedTimeAsDate = new Date(selectedDate.slice(0, -3));
+        const minimumTime = new Date(new Date().getTime() - 60*60*1000);
+        
+        if (!departureStation) {
```

Comment:
Could you put this in a method with a nice name to make it more modular and easier to skim over?